numba/pythonapi.py

```diff
@@ -26,6 +26,37 @@ def __init__(self, value, is_error=None, cleanup=None):
         self.cleanup = cleanup
 
 
+class EnvironmentManager(object):
+
+    def __init__(self, pyapi, env, env_body):
+        self.pyapi = pyapi
+        self.env = env
+        self.env_body = env_body
+
+    def add_const(self, const):
+        """
+        Add a constant to the environment, return its index.
+        """
+        # All constants are frozen inside the environment
+        if isinstance(const, str):
+            const = utils.intern(const)
```

Comment:
It just ensures that equal strings get frozen as a single constant (since we use identity testing). It would probably only matter for object mode and the call wrapper.